### PR TITLE
Add build configuration with browserlist + new CartoMap for tests

### PR DESCRIPTION
**Pull request by VictorVelarde**
### Related to https://github.com/CartoDB/carto-vl/issues/28
This PR adds configuration to Webpack so the bundling process makes transpilation using `babel-loader` and a list of supported browsers (**browserslist**). It also moves some references from dependencies to dev-dependencies.

- [x] Babel-loader config
- [x] Browserlist

---

### Related to https://github.com/CartoDB/carto-vl/issues/989 
This PR also removes carto.Map from the library. A renamed CartoMap has been created for the tests

- [x] Exclude carto.Map from bundle 
- [x] Adapt tests to new CartoMap

### CODE REVIEW

**Comment by VictorVelarde** [comment]
**browserslist** has not been defined yet with more details. 
It currently includes just `defaults`

**Comment by VictorVelarde** [comment]
Blocked by #984 (because of the `carto.Map` removal)

File: test/.eslintrc.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
     "env": {
         "mocha": true
+    },
+    "globals": {
+        "mat4": true
```
Comment (IagoLast) [inline test/.eslintrc.json]: why this is global?

Comment (VictorVelarde) [reply test/.eslintrc.json]: mat4 = gl-matrix. It is a dependency of CartoMap, and it is loaded from node_modules with `<script src="<%- glmatrix %>"></script>`

File: test/CartoMap.js [hunk 2]
```diff
@@ -1,28 +1,30 @@
-import { layerVisibility, RESOLUTION_ZOOMLEVEL_ZERO } from '../constants/layer';
-import CartoValidationError, { CartoValidationTypes as cvt } from '../errors/carto-validation-error';
-import { mat4 } from 'gl-matrix';
+// CartoMap depends on gl-matrix, which after imported is globally available as 'mat4'
+// Constants extracted from /carto-vl/src/constants/layer.js
```
Comment (elenatorro) [inline test/CartoMap.js]: This comment is not needed.

Comment (VictorVelarde) [reply test/CartoMap.js]: Do you mean the second one? the one on mat4 or both? 😄

Comment (VictorVelarde) [reply test/CartoMap.js]: Currently `carto.layerVisibility` is not available externally, nor it is `RESOLUTION_ZOOMLEVEL_ZERO`, so I'm kind of replicating them just for this utility class.

I'll keep the comments as a reminder of this.

**Pull request merged**